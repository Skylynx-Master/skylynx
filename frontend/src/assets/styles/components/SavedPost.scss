@import "../Vars.scss";

.SavedPost {
  width: 660px;
  height: 540px;
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 8% auto 0;
  background-color: $color-white;
  padding: 20px;
  text-align: center;
  border-radius: 20px;
}

.SavedPost h2 {
  font-size: 35px;
  padding: 20px 0;
  margin: 20px 0;
}

.SavedPost-check {
  width: 150px;
  height: 150px;
  margin-bottom: 20px;
}

.SavedPost p {
  font-size: 20px;
  margin-bottom: 20px;
}

.SavedPost-clock {
  width: 100px;
  height: 100px;
  margin-bottom: 20px;
}

.SavedPost a {
  font-size: 20px;
  text-decoration: underline;
}
