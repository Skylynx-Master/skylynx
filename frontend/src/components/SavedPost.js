import React from 'react';
import {Link} from 'react-router-dom'
import '../assets/styles/components/SavedPost.scss'
import checkIcon from '../assets/static/checkIcon.png';
import clockIcon from '../assets/static/clockIcon.png';


const SavedPost = () => {
  return (
    <div className="SavedPost">

      <h2>Se ha guardado su publicaciÃ³n</h2>
      <img src={checkIcon} alt="" className="SavedPost-check"></img>
      <p>Espere a que uno de nuestros asesores revise y publique su inmueble.</p>
      <img src={clockIcon} alt="clock icon" className="SavedPost-clock"></img>
      <Link to="/dashboard">Regresar a mi tablero</Link>
    </div>
  )
}

export default SavedPost;