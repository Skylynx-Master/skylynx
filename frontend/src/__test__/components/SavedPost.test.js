import React from 'react';
import SavedPost from '../../components/SavedPost';
import { mount } from 'enzyme';
import { create } from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';

describe("SavedPost Component Testing", () => {
  const savepost = mount(
    <BrowserRouter>
      <SavedPost />
    </BrowserRouter>
  );
  test("SavedPost Render", () => {
    expect(savepost.length).toEqual(1);
  });

  test("SavedPost container render", () => {
    expect(savepost.find(".SavedPost").length).toEqual(1);
  });

  test("SavedPost title Render", () => {
    expect(savepost.find(".SavedPost-title").length).toEqual(1);
  });

  test("SavedPost image render", () => {
    expect(savepost.find(".SavedPost-check").length).toEqual(1);
  });

  test("SavedPost instructions render", () => {
    expect(savepost.find(".Savedpost-instructions").length).toEqual(1);
  });

  test("SavedPost instructions text", () => {
    expect(savepost.find(".Savedpost-instructions").text()).toEqual(
      "Espere a que uno de nuestros asesores revise y publique su inmueble."
    );
  });

  test("SavedPost second image render", () => {
    expect(savepost.find(".SavedPost-clock").length).toEqual(1);
  });

  test("SavedPost link render", () => {
    expect(savepost.find(".SavedPost-route").length).toEqual(3);
  });

});

describe("SavedPost Snapshot", () => {
  test("Test UI SavedPost component", () => {
    const publish = create(
      <BrowserRouter>
        <SavedPost />
      </BrowserRouter>
    );
    expect(publish.toJSON()).toMatchSnapshot();
  });
});