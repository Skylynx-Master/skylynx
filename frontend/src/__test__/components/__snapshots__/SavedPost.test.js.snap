// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedPost Snapshot Test UI SavedPost component 1`] = `
<div
  className="SavedPost"
>
  <h2
    className="SavedPost-title"
  >
    Se ha guardado su publicaci√≥n
  </h2>
  <img
    alt=""
    className="SavedPost-check"
    src=""
  />
  <p
    className="Savedpost-instructions"
  >
    Espere a que uno de nuestros asesores revise y publique su inmueble.
  </p>
  <img
    alt="clock icon"
    className="SavedPost-clock"
    src=""
  />
  <a
    className="SavedPost-route"
    href="/dashboard"
    onClick={[Function]}
  >
    Regresar a mi tablero
  </a>
</div>
`;
